Seminars/Sem.02/Pract.02/Task07.cpp

```diff
@@ -0,0 +1,177 @@
+//Gabriela Mladenova
+//SI
+//Group 1
+//FN: 3MI0600225
+
+#include <iostream>
+#include <fstream>
+
+const int SIZE = 128;
+const int USERS = 100;
+const char FILENAME1[] = "usersList.txt";
+
+struct User
+{
+    char name[SIZE + 1];
+    char email[SIZE + 1];
+    char password[SIZE + 1];
+};
+
+struct System
+{
+    User systemOfUsers[USERS];
+    int countOfStudentsInSystem;
+};
+
+void inputUsers(System &newSystem)
+{
+    std::ofstream writeInFile(FILENAME1, std::ios::app);
+    if (!writeInFile.is_open())
+    {
+        std::cout << "ERROR! The file could not be opened!";
+        return;
+    }
+
+    for (int i = 0; i < USERS; ++i)
+    {
+        writeInFile << newSystem.systemOfUsers[i].name
+            << newSystem.systemOfUsers[i].email 
+            << newSystem.systemOfUsers[i].password << std::endl;
+
+        ++newSystem.countOfStudentsInSystem;
+    }
+
+    writeInFile.close();
+}
+
+void printUsersList(System &newSystem)
+{
+    std::ifstream readFromFile(FILENAME1);
+    if (!readFromFile.is_open())
+    {
+        std::cout << "ERROR! The file could not be opened!";
+        return;
+    }
+
+    int i;
+    for (i = 0; i < USERS; ++i)
+    {
+        readFromFile >> newSystem.systemOfUsers[i].name
+            >> newSystem.systemOfUsers[i].email
+            >> newSystem.systemOfUsers[i].password;
+    }
+    newSystem.countOfStudentsInSystem = i;
```

Comment:
The idea is to store the current number of users in countOfStudentsInSystem, not the maximum possible size